lib/elastomer/client/docs.rb

```diff
@@ -261,6 +261,21 @@ def delete_by_query(query, params = nil)
         @client.delete_by_query(query, update_params(params))
       end
 
+      # Matches a document to the queries stored in the percolator for the given
+      # index. Type is required.
+      #
+      # Examples
+      #
+      #   index.register_percolator_query 1, { :query => { :match => { :author => "pea53" } } }
+      #   index.docs(type).percolate({ :doc => { :author => "pea53" } })
+      #
+      # Returns the response body as a Hash
+      def percolate(body, params = {})
+        raise 'percolate requires document type' if type.nil?
```

Comment:
`:type` can be passed in as a parameter so we don't know for sure if it's missing yet. I believe we normally allow the URI template to catch missing path segments.